libc/include/llvm-libc-macros/float16-macros.h

```diff
@@ -13,7 +13,8 @@
 
 #if defined(__FLT16_MANT_DIG__) &&                                             \
     (!defined(__GNUC__) || __GNUC__ >= 13 || defined(__clang__)) &&            \
-    !defined(__arm__) && !defined(_M_ARM) && !defined(__riscv)
+    !defined(__arm__) && !defined(_M_ARM) && !defined(__riscv) &&              \
```

Comment:
notice that msvc arm64 define __ARM_ARCH rather than these __arm__ macros. It is not a concern for this patch but may cause problem later on

Reply:
AArch64 is the CPU target with the best `float16` support. Its equivalent of the x86-64 F16C extension is part of the base Armv8-A ISA and mandatory, so we never had issues with compiler runtime soft-float conversion functions on AArch64, since they're not used there. We only had issues on 32-bit Arm, and MSVC defines `_M_ARM` when targeting 32-bit Arm according to https://learn.microsoft.com/en-us/cpp/preprocessor/predefined-macros?view=msvc-170.